In [2]:
import pandas as pd

df = pd.read_csv('../data/student_dropout.csv')

In [3]:
# Basic Dataset Overview
df.shape
df.columns
df.head()
df.tail()

Marital status  Application mode  Application order  Course  \
4419               1                 1                  6      15   
4420               1                 1                  2      15   
4421               1                 1                  1      12   
4422               1                 1                  1       9   
4423               1                 5                  1      15   

      Daytime/evening attendance  Previous qualification  Nacionality  \
4419                           1                       1            1   
4420                           1                       1           19   
4421                           1                       1            1   
4422                           1                       1            1   
4423                           1                       1            9   

      Mother's qualification  Father's qualification  Mother's occupation  \
4419                       1                       1                    6   
4420                       1                       1                   10   
4421                      22                      27                   10   
4422                      22                      27                    8   
4423                      23                      27                    6   

      ...  Curricular units 2nd sem (credited)  \
4419  ...                                    0   
4420  ...                                    0   
4421  ...                                    0   
4422  ...                                    0   
4423  ...                                    0   

      Curricular units 2nd sem (enrolled)  \
4419                                    6   
4420                                    6   
4421                                    8   
4422                                    5   
4423                                    6   

      Curricular units 2nd sem (evaluations)  \
4419                                       8   
4420                                       6   
4421                                       9   
4422                                       6   
4423                                       6   

      Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
4419                                    5                         12.666667   
4420                                    2                         11.000000   
4421                                    1                         13.500000   
4422                                    5                         12.000000   
4423                                    6                         13.000000   

      Curricular units 2nd sem (without evaluations)  Unemployment rate  \
4419                                               0               15.5   
4420                                               0               11.1   
4421                                               0               13.9   
4422                                               0                9.4   
4423                                               0               12.7   

      Inflation rate   GDP    Target  
4419             2.8 -4.06  Graduate  
4420             0.6  2.02   Dropout  
4421            -0.3  0.79   Dropout  
4422            -0.8 -3.12  Graduate  
4423             3.7 -1.70  Graduate  

[5 rows x 35 columns]

In [4]:
# Data Quality Checks
df.info()
df.describe(include='all')  # to see categorical stats
df.isnull().sum()
df.duplicated().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 35 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance                      4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Nacionality                                     4424 non-null   int64  
 7   Mother's qualification                          4424 non-null   int64  
 8   Father's qualification                          4424 non-null   int64  
 9   Mother's occupation                      

np.int64(0)

In [7]:
# Target Column Exploration
df['Target'].value_counts()
df['Target'].value_counts(normalize=True)

Target
Graduate    0.499322
Dropout     0.321203
Enrolled    0.179476
Name: proportion, dtype: float64

# 🧾 Day 1 Summary – Initial Data Exploration

### 📊 Dataset Overview
- Total records: `5`
- Total features: `35`
- Source: Kaggle – Predict Student Dropout and Academic Success
- Dataset includes demographic, academic, and socioeconomic variables

---

### 🎯 Target Variable: `Target`
- **Graduate**: 49.9%
- **Dropout**: 32.1%
- **Enrolled**: 17.9%
- Class imbalance is present but not extreme
- Suitable for **multi-class classification** or **binary dropout prediction**

---

### 🧼 Data Quality
- Missing values: Use `df.isnull().sum()` to identify specific fields
- Duplicates: `df.duplicated().sum()` (should be near zero)
- Categorical columns like `'Marital status'`, `'Application mode'`, and `'Gender'` will require encoding
- Numerical columns appear clean and ready for EDA

---

### ✅ Next Steps (Planned for Day 2)
- Visualize target class distribution
- Explore correlations between key academic variables and dropout
- Start preprocessing plan: encoding, feature selection, imbalance handling

---

### 💾 Status
- Dataset loaded and verified
- Basic structure and target classes understood
- Notebook saved to `notebooks/eda.ipynb`